Python/Source/Range.cpp

```diff
@@ -0,0 +1,41 @@
+#include <type_traits>
+#include <variant>
+#include "Range.hpp"
+
+using namespace pybind11;
+using namespace Rover;
+
+namespace {
+  class PythonRange {
+    public:
+      using Type = Range<object, object>::Type;
+      static_assert(std::is_same_v<Type, Range<object, object, object>::Type>);
+
+      PythonRange(object begin, object end)
+        : m_impl(Range(std::move(begin), std::move(end))) {}
+
+      PythonRange(object begin, object end, object granularity)
+        : m_impl(Range(std::move(begin), std::move(end),
+            std::move(granularity))) {}
+
+      Type generate(Evaluator& e) {
+        return std::visit([&] (auto& impl) {
+          return impl.generate(e);
+        }, m_impl);
+      }
+
+    private:
+      using ContinuousRange = Range<object, object>;
+      using DiscreteRange = Range<object, object, object>;
+      std::variant<ContinuousRange, DiscreteRange> m_impl;
+  };
+}
+void Rover::export_range(module& module) {
+  class_<PythonRange>(module, "Range")
+    .def(init<object, object>())
+    .def(init<object, object, object>())
+    .def("generate", &PythonRange::generate);
+
+  implicitly_convertible<PythonRange, Box<PythonRange::Type>>();
+  implicitly_convertible<Box<PythonRange::Type>, PythonRange>();
+}
```

Comment:
Add a conversion from ```PythonRange``` to ```Box<object>```.

Reply:
PythonRange::Type is always object, so it is the same thing. Contracted that to object anyway.